sno/gpkg.py

```diff
@@ -234,6 +241,76 @@ def geom_to_ogr(gpkg_geom, parse_srs=False):
     return geom
 
 
+def ogr_to_geom(ogr_geom):
+    """
+    Given an OGR geometry object,
+    construct a GPKG geometry value.
+    http://www.geopackage.org/spec/#gpb_format
+
+    Arbitrarily, this only produces little-endian geometries.
+    """
+    wkb = ogr_geom.ExportToWkb()
+
+    # always produce little endian
+    flags = 1
+    ogr_geom_type = ogr_geom.GetGeometryType()
+
+    # magic number and version
+    pieces = [b'GP\x00']
+
+    # Flags
+    empty = ogr_geom.IsEmpty()
+    has_z = ogr.GT_HasZ(ogr_geom_type)
+    if empty:
+        # no envelope.
+        flags |= 0x10
+    elif has_z:
+        # XYZ envelope
+        flags |= 0x04
+    else:
+        # XY envelope
+        flags |= 0x02
+    pieces.append(struct.pack('<B', flags))
+
+    # srs_id
+    srid = 0
+    srs = ogr_geom.GetSpatialReference()
+    if srs:
+        srs.AutoIdentifyEPSG()
+        if srs.IsProjected():
+            srid = int(srs.GetAuthorityCode("PROJCS"))
+        elif srs.IsGeographic():
+            srid = int(srs.GetAuthorityCode("GEOGCS"))
+    pieces.append(struct.pack('<i', srid))
+
+    # TODO: XYZM/XYM envelope.
```

Comment:
ok. Fine with me. what are we replacing them with?

Looks like the `geom_envelope` function below looks like it can be used for spatial indexing, and efficiently loads these from GPKG geometries when they're available. So removing them will cause _some_ kind of performance hit in repositories generated from patches.